### PR TITLE
(feat)support for backward interval generation

### LINKED ISSUE #2509: Support backward interval generation in IN operator
### Is your feature request related to a problem?

IN operator support interval generation:
```
select timestamp, price
from trades
where timestamp in '2018-01-01T13:00:00;1h;24h;10000'
```

The above query would return results from 13:00-14:00 intervals for 10K days starting with 2018-01-01, so the results are in future when compared with the start date.

Would be nice to support backward iteration syntax, e.g. '2018-01-01T13:00:00;1h;-24h;10000', which would return intervals in the past.

### Describe the solution you'd like.

_No response_

### Describe alternatives you've considered.

_No response_

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by anonymous-baaka**
closes #2509 

supports backward interval generation for second, min, hour, month, year.
example: 
select * from 'test-quotes-big.csv' where ts in '1971-11-02T00:00:00.000000Z;1h;-6d;10'

would generate 10 rows starting from the specified timestamp into the past each with a 6 days interval

image

### CODE REVIEW

**Comment by bziobrowski** [comment]
Hello @anonymous-baaka . 
Please add tests showing that backward intervals work .

File: core/src/main/java/io/questdb/griffin/model/IntervalUtils.java [hunk 1]
```diff
@@ -534,6 +536,22 @@ public static void parseIntervalEx(CharSequence seq, int lo, int lim, int positi
             }
         }
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/model/IntervalUtils.java:537]: Please reformat code using qdb setttings . 
IntelliJ should pick up the config automatically but if you don't have reformat & rearrange turn on in 'Actions on save' then you've to select IntervalUtils and choose Code | Reformat File on .

**Comment by anonymous-baaka** [comment]
@bziobrowski 
> Please add tests showing that backward intervals work

By this, do you mean I should create new test cases/ test modules to verify if it works or that I should just add the results of existing test modules?

**Comment by bluestreak01** [comment]
these should be new tests :)

**Comment by bluestreak01** [comment]
feel free to re-open PR when ready to continue :)

**Pull request closed without merging**